viz.md

```diff
@@ -58,13 +60,23 @@ def viz_selection():
     sequence_length = 5e4
     sweep_model = msprime.SweepGenicSelection(position=sequence_length/2,
     s=0.01, start_frequency=0.5e-4, end_frequency=0.99, dt=1e-6)
-    ts_selection = msprime.sim_ancestry(9,
+    ts_selection = msprime.sim_ancestry(12,
+        ploidy=1,
         model=[sweep_model, msprime.StandardCoalescent()],
         population_size=1e4,
-        recombination_rate=1e-8,
+        recombination_rate=2e-8,
         sequence_length=sequence_length,
-        random_seed=9,
+        random_seed=222,
     )
+    tables = ts_selection.dump_tables()
+    tables.nodes.metadata_schema = tskit.MetadataSchema.permissive_json()
+    node_names = [{}] * tables.nodes.num_rows
```

Comment:
This is quite a complicated way to do it - why not:
```
tables.nodes.clear()
tables.nodes.metadata_schema = ...
for node in ts.nodes():
      metadata = {}
      if node.is_sample():
           metadata["name"] = ...
      tables.nodes.add_row(node.replace(metadata=metadata))
```
It won't be much slower (we're doing a bunch of stuff in Python either way) and it's a lot easier to follow.

Reply:
Sure, that's fine. I'm hoping most people won't be reading this bit of the code anyway (it's hidden, and there was a previous push motivated by Ben Haller to make all the example code at the end self-contained, so the reader doesn't need to look at the hidden stuff). As you say, it's best to make it clear to ourselves even in the hidden stuff though.